### DISCUSSION TITLE
Filter by cell tags

### DISCUSSION THREAD

**Discussion by ig005bvr** [milaboratory/mixcr#1556] [Q&A] [answered]
How can I filter my dataset based on specific cell tags derived from gene expression data? I've acquired a list of cell tags from Parse genome expression data and need to filter VDJ reads to keep only alignments from the target cells. The cell tags are formatted as "01_02_03", and it seems they cannot be directly imported as a whitelist. The step should be somewhere after  transformation:
            type: concatenate
            separator: '_'
            sources:
              - type: tag
                name: CELL1CONVERTED
              - type: tag
                name: CELL2CONVERTED
              - type: tag
                name: CELL3CONVERTED
I can supply a csv/txt with target tags. Thank you in advance!

**Comment by mizraelson** [comment, marked as answer]
Hi, currently there is no such feature to filter certain cells. We will add it soon. For now recommend just exporting all cells first and then manually filter by the CELL tags using Python/R for example.

**Reply by ig005bvr** [reply]
Thank you! What settings would you recommend for keeping all or at least one alignment per all CELL barcodes that were identified at the refine tags and sort and assemble steps? For my protocol, I use generic-ht-single-cell-amplicon-with-umi. How do you keep all reads and not discard them?

**Reply by mizraelson** [reply]
Hi, MiXCR utilizes a variety of filters and thresholds, which can result in some cells identified early in the process not appearing in the final output. This approach ensures the delivery of the most reliable results. Essentially, if certain alignments, UMIs, or cells were discarded, it likely indicates there were contamination or quality issues with them. If you're able to share your data, I can provide specific insights into why a particular cell was discarded. However, without looking at the actual data, it's important to note that there isn't a single parameter that can preserve all alignments without affecting the quality of the output.
